server/graphql/general/src/sync_api_error.rs

```diff
@@ -0,0 +1,312 @@
+use async_graphql::*;
+use graphql_core::standard_graphql_error::StandardGraphqlError;
+use repository::SyncLogRowErrorCode;
+use service::sync::{
+    api::{SyncApiError, SyncApiErrorVariant, SyncApiV5CreatingError, SyncErrorCodeV5},
+    site_info::RequestAndSetSiteInfoError,
+    sync_status::SyncLogError,
+};
+use util::format_error;
+
+#[derive(SimpleObject)]
+pub struct SetSyncSettingErrorNode {
+    pub error: SyncErrorInterface,
+}
+
+impl SetSyncSettingErrorNode {
+    pub fn map_error(error: RequestAndSetSiteInfoError) -> Result<Self> {
+        use RequestAndSetSiteInfoError as from;
+
+        let error = match &error {
+            // Structured error
+            from::RequestSiteInfoError(api_error) => {
+                SyncErrorInterface::from_sync_api_error(api_error)
+            }
+            from::SiteUUIDIsBeingChanged(_, _) => {
+                SyncErrorInterface::from_variant(SyncErrorVariant::SiteUUIDIsBeingChanged, &error)
+            }
+            from::SyncApiV5CreatingError(SyncApiV5CreatingError::CannotParseSyncUrl(_, _)) => {
+                SyncErrorInterface::from_variant(SyncErrorVariant::InvalidUrl, &error)
+            }
+            // Standard Graphql Errors
+            _ => return Err(StandardGraphqlError::from_error(&error)),
+        };
+
+        Ok(Self { error })
+    }
+}
+
+#[derive(Interface)]
+#[graphql(name = "SyncErrorInterface")]
+#[graphql(field(name = "description", type = "String"))]
+#[graphql(field(name = "full_error", type = "String"))]
+pub enum SyncErrorInterface {
+    MappedSyncError(MappedSyncError),
```

Comment:
Was wondering how this is used by a client. In the `FullSyncStatusNode` the error is an option: 
```
    error: Option<SyncErrorInterface>,
```

so we can check to see if it has an error.. but then we need to check both properties to see what the error is? Wouldn't it be simpler to have one of the enum values of `SyncErrorVariant` being `Unknown` and have the `error_variant` top level along with `full_error`?

Reply:
Wow 🙏 🙏 🙏 I completely overcooked, i tried following the pattern, this is definitely the case where it's much easier to diverge. be8fdeceb048ea7a54aa049a8c655a4490f5ed8c (straight into head)

So note to self, use the error union variant only when they return more complex and different types (based on the error), otherwise just an enum variant does the job much simpler